pkg/steps/pod_test.go

```diff
@@ -182,3 +184,135 @@ func TestPodStepExecution(t *testing.T) {
 		})
 	}
 }
+
+func TestGetPodObject(t *testing.T) {
+	testCases := []struct {
+		name        string
+		podStep     func(*podStep)
+		expectedPod func(*v1.Pod)
```

Comment:
This might be easier to read as a `func(*v1.Pod, *testing.T)` and just have this func validate the pod itself

Reply:
Also technically you only validate the podspec, right?

Reply:
We validate podSpec with PodTemplate which is mutated manually in the test scenario. So it validated function generated podSpec with our template. I think the test will become more complex if I stick with one function.

Reply:
I still need to "construct reference point" so second function is needed. I could remove all `func expectedPodTemplate() *v1.Pod` but this means this test will become unextendable and very narrow. With current implementation it can be extended to test any podSpec mutation.

Reply:
Instead of taking a template, mutating it and doing a DeepEqual, just validate the fields you need. Don't worry about this not being extensible, infinitely extensible unit tests aren't something we really need to be worrying about